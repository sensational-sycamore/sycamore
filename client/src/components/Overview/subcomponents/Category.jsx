import React from 'react';

const Category = ({ product }) => (
  <div>
    <h2>{product.category}</h2>
  </div>
);

export default Category;
