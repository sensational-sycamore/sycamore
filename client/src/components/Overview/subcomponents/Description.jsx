import React from 'react';

const Description = ({ product }) => (
  <div>
    <h1>Description</h1>
    <div>{product.slogan}</div>
    <div>{product.description}</div>
  </div>
);

export default Description;