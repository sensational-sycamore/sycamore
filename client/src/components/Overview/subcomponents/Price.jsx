import React from 'react';
import css from '../styles/price.module.css';

const Price = ({ currStyle }) => {
  const whichPrice = () => {
    if (currStyle.sale_price) {
      return (
        <div>
          <div className={css.oldPrice}>${ currStyle.original_price }</div>
          <div className={css.newPrice}>${currStyle.sale_price}</div>
        </div>
      );
    } else {
      return (
        <div className={css.regularPrice}>${ currStyle.original_price }</div>
      );
    }
  };

  return whichPrice();
};


export default Price;
