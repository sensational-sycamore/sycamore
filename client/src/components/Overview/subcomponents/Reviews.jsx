import React from 'react';
import css from '../styles/ratings.modules.css';


const Reviews = () => {
  return (
    <div>
      <a className={css.reviewLink} href="#ratings-and-reviews">Read all reviews</a>
    </div>
  );
};

export default Reviews;
