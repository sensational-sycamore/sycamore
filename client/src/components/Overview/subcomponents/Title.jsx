import React from 'react';
import css from '../styles/title.modules.css';

const Title = ({ name }) => (
  <div>
    <h1 className={css.title}>{name}</h1>
  </div>
);

export default Title;
