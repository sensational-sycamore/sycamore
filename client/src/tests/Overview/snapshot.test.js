//snapshot testing

import React from 'react';
import renderer from 'react-test-renderer';
import Title from '../../components/Overview/subcomponents/Title.jsx';
import Category from '../../components/Overview/subcomponents/Category.jsx';
import Description from '../../components/Overview/subcomponents/Description.jsx';
import Reviews from '../../components/Overview/subcomponents/Reviews.jsx';
import Price from '../../components/Overview/subcomponents/Price.jsx';


// fake product object
const fakeProductDetails = {
  'id': 63618,
  'campus': 'hr-sfo',
  'name': 'Infinity Stone',
  'slogan': 'Reality is often disappointing. That is, it was. Now, reality can be whatever I want.',
  'description': 'The Infinity Stones are six immensely powerful stone-like objects tied to different aspects of the universe, created by the Cosmic Entities. Each of the stones possesses unique capabilities that have been enhanced and altered by various alien civilizations for millennia.',
  'category': 'Accessories',
  'default_price': '50000000.00',
  'created_at': '2021-12-21T17:19:40.556Z',
  'updated_at': '2021-12-21T17:19:40.556Z',
  'features': []
};

it('Title component renders correctly', () => {
  const tree = renderer
    .create(<Title product={fakeProductDetails}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it('Category component renders correctly', () => {
  const tree = renderer
    .create(<Category product={fakeProductDetails}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it('Description component renders correctly', () => {
  const tree = renderer
    .create(<Description product={fakeProductDetails}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it('Reviews component renders correctly', () => {
  const tree = renderer
    .create(<Reviews product={fakeProductDetails}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it('Price component renders correctly', () => {
  const tree = renderer
    .create(<Price product={fakeProductDetails}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
