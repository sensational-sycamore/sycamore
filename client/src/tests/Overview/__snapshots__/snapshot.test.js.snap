// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category component renders correctly 1`] = `
<p>
  Accessories
</p>
`;

exports[`Description component renders correctly 1`] = `
<div>
  <h3>
    Reality is often disappointing. That is, it was. Now, reality can be whatever I want.
  </h3>
  <h3>
    The Infinity Stones are six immensely powerful stone-like objects tied to different aspects of the universe, created by the Cosmic Entities. Each of the stones possesses unique capabilities that have been enhanced and altered by various alien civilizations for millennia.
  </h3>
</div>
`;

exports[`Reviews component renders correctly 1`] = `
<div>
  <a
    href="#ratings-and-reviews"
  >
    Read all reviews
  </a>
</div>
`;

exports[`Title component renders correctly 1`] = `
<div>
  <h1>
    Infinity Stone
  </h1>
</div>
`;
